packages/upload-api/test/delegations-storage-tests.js

```diff
@@ -0,0 +1,91 @@
+import * as assert from 'node:assert'
+import * as principal from '@ucanto/principal'
+import * as Ucanto from '@ucanto/interface'
+import * as ucanto from '@ucanto/core'
+import { createSampleDelegation } from '../src/utils/ucan.js'
+import * as Types from '../src/types.js'
+
+/**
+ * @param {object} [opts]
+ * @param {Ucanto.Signer<Ucanto.DID>} [opts.issuer]
+ * @param {Ucanto.Principal} [opts.audience]
+ * @param {Ucanto.Capabilities} [opts.capabilities]
+ * @returns {Promise<Ucanto.Delegation>}
+ */
+async function createDelegation(opts = {}) {
+  const {
+    issuer = await principal.ed25519.generate(),
+    audience = issuer,
```

Comment:
`issuer` and `audience` are always passed so this seems like unnecessary generalization.

Reply:
hm I'm not sure what you mean by this, could you expand on this?